kafka-client/src/main/java/dev/responsive/utils/Iterators.java

```diff
@@ -284,6 +291,42 @@ public T next() {
     }
   }
 
+  private static class KeyMappingIterator<I, O, V> implements KeyValueIterator<O, V> {
```

Comment:
not used - I used this in the original implementation when I was client-side filtering but dropped that approach. Either way, this is a worthwhile utility iterator so I'm keeping it